### PR TITLE
Refactor path_extract_config

### PR DESCRIPTION
**Pull request by ilkilic**
Retrieve the ExtractionTargetsConfiguration from the relevant ``run`` folder, if available, rather than from the root directory, which is subject to change.

### CODE REVIEW

**Comment by AurelienJaquier** [comment]
Could you add an explanation about that in the docstring of emodel_settings under path_extract_config? That the behavior will change depending on the emodel_dir value?

**Comment by ilkilic** [comment]
Actually, self.emodel_dir will never be `None` (https://github.com/BlueBrain/BluePyEModel/blob/31dc58b30fd62f8a4b4b0810acfa3dc1f16d9e65/bluepyemodel/access_point/local.py#L115)
Therefore, I simplified the code by eliminating the unnecessary 'if self.emodel_dir:' checks

File: bluepyemodel/access_point/local.py [hunk 1]
```diff
@@ -345,13 +347,10 @@ def store_model_configuration(self, configuration, path=None):
...
         """Return the path to the directory containing the mechanisms for the current emodel"""
 
-        if self.emodel_dir:
-            mechanisms_directory = self.emodel_dir / "mechanisms"
-        else:
-            mechanisms_directory = Path("./") / "mechanisms"
+        mechanisms_directory = self.emodel_dir / "mechanisms"
 
         if mechanisms_directory.is_dir():
-            return mechanisms_directory.resolve()
```
Comment (AurelienJaquier) [inline bluepyemodel/access_point/local.py:354]: why remove the .resolve() here?

Comment (AurelienJaquier) [reply bluepyemodel/access_point/local.py:354]: Technically, in the improbable case where the user gives an emodel_dir, the path is not absolute I think. But yes, anyway, it does not harm to keep it ;)

Comment (ilkilic) [reply bluepyemodel/access_point/local.py:354]: You are correct, it might not be absolute in that case, yes let's keep it :)

**Pull request merged**